core/dynamo.c

```diff
@@ -1444,14 +1472,20 @@ dynamo_process_exit(void)
          */
         instrument_exit();
 
-# ifdef CLIENT_INTERFACE
+# ifdef CLIENT_SIDELINE
+        /* We only need do a second synch-all if there are sideline client threads. */
```

Comment:
Ditto: how about a num threads check to avoid all the locks when the common case has zero sideline